### DISCUSSION TITLE
Has the UUID global function for Script Kit changed?

### DISCUSSION THREAD

**Discussion by lannonbr** [johnlindquist/kit#681] [Q&A] [answered]
I have a script that I can press `opt+u` and have a UUID copied to my clipboard. Here's the script:

```js
copy(uuid())
```

That said, when I run it I expect a string like this:

```
9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dcb6d
```

but as of late I have something like this appear:

```
qVFutxhczd2Dza2FjIARb
```

Which seems to be another unique ID string, but not compliant to a RFC4122 UUID. Any idea on what is happening?

**Comment by johnlindquist** [comment, marked as answer]
Yes, the uuid() function changed:

Kit switch from using the "uuid" package to [https://www.npmjs.com/package/nanoid](https://www.npmjs.com/package/nanoid)

https://github.com/johnlindquist/globals/blob/main/src/nanoid.ts#L3

### Why nanoid

When Script Kit upgraded to node 17, webpack (during Kit app builds) started throwing this SSL error:
https://github.com/webpack/webpack/issues/14532

I can't remember exactly how/why `uuid` was triggering it, sorry, but I figured it was simple enough to just use another common library. I overlooked that it wasn't RFC4122 UUID compliant.

### Expectations?

Are your expectations that "uuid" means "RFC4122 UUID compliant"? I'm far from an expert in that field and if so, I'm happy to look into changing it. I'm very open to suggestions.

**Reply by lannonbr** [reply]
Yeah I think if I expect to see a function titled `uuid` that it would be compliant to RFC4122. I don't mind the new library, but maybe either change the global to `global.nanoid` or `global.randomId` or something similar, or maybe have an explicit post in the discussions saying that the underlying change was a breaking change as if people were dependent on a RFC4122 compliant UUID this would break their software with such.

**Reply by johnlindquist** [reply]
How about I switch `uuid` to using the build-in crypto.randomUUID? https://nodejs.org/api/crypto.html#cryptorandomuuidoptions

My gut feeling is that would align with most people's expectations.
